core/src/main/java/com/segment/analytics/kotlin/core/platform/Plugin.kt

```diff
@@ -70,6 +70,14 @@ interface EventPlugin : Plugin {
         return payload
     }
 
+    override fun execute(event: BaseEvent): BaseEvent? = when (event) {
```

Comment:
should this be marked `final`?

Reply:
no. we shouldn't, because we extends `execute` in edge function. amplitude also overrides this method.